src/main/java/net/dv8tion/jda/core/requests/impl/WebhookAction.java

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.core.requests.impl;
+
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.entities.EntityBuilder;
+import net.dv8tion.jda.core.entities.Icon;
+import net.dv8tion.jda.core.entities.Webhook;
+import net.dv8tion.jda.core.exceptions.RateLimitedException;
+import net.dv8tion.jda.core.requests.Request;
+import net.dv8tion.jda.core.requests.Response;
+import net.dv8tion.jda.core.requests.RestAction;
+import net.dv8tion.jda.core.requests.Route;
+import org.json.JSONObject;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+public class WebhookAction extends RestAction<Webhook>
+{
+
+    protected String name;
+    protected Icon avatar = null;
+
+    public WebhookAction(JDA api, Route.CompiledRoute route, String name)
+    {
+        super(api, route, null);
+        this.name = name;
+    }
+
+    /**
+     * Sets the <b>Name</b> for the custom Webhook User
+     *
+     * @param  name
+     *         A not-null String name for the new Webhook user.
+     *
+     * @throws IllegalArgumentException
+     *         If the specified name is not in the range of 2-100.
+     *
+     * @return The current WebhookAction for chaining convenience.
+     */
+    public WebhookAction setName(String name)
+    {
+        if (name == null || name.length() < 2 || name.length() > 100)
+            throw new IllegalArgumentException("The webhook name must be in the range of 2-100!");
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Sets the <b>Avatar</b> for the custom Webhook User
+     *
+     * @param  icon
+     *         An {@link net.dv8tion.jda.core.entities.Icon Icon} for the new avatar.
+     *         Or null to use default avatar.
+     *
+     * @return The current WebhookAction for chaining convenience.
+     */
+    public WebhookAction setAvatar(Icon icon)
+    {
+        this.avatar = icon;
+        return this;
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    public void queue(Consumer<Webhook> success, Consumer<Throwable> failure)
+    {
+        this.setData();
+        super.queue(success, failure);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Webhook block() throws RateLimitedException
+    {
+        this.setData();
+        return super.block();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Webhook block(long timeout, TimeUnit timeUnit) throws TimeoutException, RateLimitedException
+    {
+        this.setData();
+        return super.block(timeout, timeUnit);
+    }
+
+    private void setData()
+    {
+        JSONObject data = new JSONObject();
+        data.put("name",   name);
```

Comment:
Name is the only required field for this endpoint and I do set the name upon creating the WebhookAction as you can see [here](https://github.com/DV8FromTheWorld/JDA/pull/205/files#diff-8de0bd2db2ba4be8271e3647aba58ad5R1001) and pass it to the constructor.